src/System.Drawing.Common/tests/mono/System.Imaging/MetafileTest.cs

```diff
@@ -0,0 +1,705 @@
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Static_GetMetafileHeader_IntPtr()
+        {
+            string filename = Helpers.GetTestBitmapPath(MetafileTest.WmfPlaceable);
+            using (Metafile mf = new Metafile(filename))
+            {
+
+                IntPtr hemf = mf.GetHenhmetafile();
+                Assert.True(hemf != IntPtr.Zero);
+
+                Assert.Throws<ArgumentException>(() => Metafile.GetMetafileHeader(hemf));
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_IntPtrBool_Zero()
+        {
+            Assert.Throws<ArgumentException>(() => new Metafile(IntPtr.Zero, false));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_IntPtrEmfType_Zero()
+        {
+            Assert.Throws<ArgumentException>(() => new Metafile(IntPtr.Zero, EmfType.EmfOnly));
+        }
+
+        private void CheckEmptyHeader(Metafile mf, EmfType type)
+        {
+            MetafileHeader mh = mf.GetMetafileHeader();
+            Assert.Equal(0, mh.Bounds.X);
+            Assert.Equal(0, mh.Bounds.Y);
+            Assert.Equal(0, mh.Bounds.Width);
+            Assert.Equal(0, mh.Bounds.Height);
+            Assert.Equal(0, mh.MetafileSize);
+            switch (type)
+            {
+                case EmfType.EmfOnly:
+                    Assert.Equal(MetafileType.Emf, mh.Type);
+                    break;
+                case EmfType.EmfPlusDual:
+                    Assert.Equal(MetafileType.EmfPlusDual, mh.Type);
+                    break;
+                case EmfType.EmfPlusOnly:
+                    Assert.Equal(MetafileType.EmfPlusOnly, mh.Type);
+                    break;
+                default:
+                    Assert.True(false, string.Format("Unknown EmfType '{0}'", type));
+                    break;
+            }
+        }
+
+        private void Metafile_IntPtrEmfType(EmfType type)
+        {
+            using (Bitmap bmp = new Bitmap(10, 10, PixelFormat.Format32bppArgb))
+            {
+                using (Graphics g = Graphics.FromImage(bmp))
+                {
+                    IntPtr hdc = g.GetHdc();
+                    try
+                    {
+                        Metafile mf = new Metafile(hdc, type);
+                        CheckEmptyHeader(mf, type);
+                    }
+                    finally
+                    {
+                        g.ReleaseHdc(hdc);
+                    }
+                }
+            }
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_IntPtrEmfType_Invalid()
+        {
+            Assert.Throws<ArgumentException>(() => Metafile_IntPtrEmfType((EmfType)Int32.MinValue));
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void Metafile_IntPtrEmfType_EmfOnly()
+        {
```

Comment:
Ditto about being disabled. The other two are this same method.

Reply:
fyi, they were disabled because they fail because they include this line:
```
var bufferGraphics = Graphics.FromHwndInternal(IntPtr.Zero)
```

This will call into `System.Drawing.LibX11Functions.XOpenDisplay(IntPtr display)` which takes a dependency on X11, which is not enabled on the build servers.

I'm about to replace this by:

```
using (Bitmap bmp = new Bitmap(10, 10, PixelFormat.Format32bppArgb))
using (Graphics g = Graphics.FromImage(bmp))
```

which should work on headless Linux as well.

Increasing coverage, one test a time ;-)

Reply:
Thanks for fixing that!